content/en/real_user_monitoring/guide/monitor-your-nextjs-app-with-rum.md

```diff
@@ -0,0 +1,211 @@
+---
+title: Monitor Your Next.js App With RUM
+kind: guide
+description: Guide for monitoring Next.js applications with RUM.
+further_reading:
+- link: '/monitors/create/types/real_user_monitoring/'
+  tag: 'Documentation'
+  text: 'Learn about RUM Monitors'
+---
+
+## Overview
+[Next.js][1] is a JavaScript framework that is used to create React.js web pages and Node.js APIs. You can integrate Next.js with RUM to monitor your frontend applications for browser-related metrics that you give insight into performance and user behavior.
+
+## Setup
+
+Follow the steps below to set up Datadog RUM browser monitoring.
+
+### Create a new application
+
+1. Navigate to **[UX Monitoring > Real User Monitoring][2]**.
+2. Click the **New Application** button.
+3. Make sure JS is selected, then enter a name for your application and click **Create New RUM Application**. A `clientToken` and `applicationId` are automatically generated for your application.
+
+### Instrument your application
+
+1. Choose your instrumentation type, then copy and paste the code snippet from the Datadog RUM UI into the appropriate file based on the instrumentation type.
+
+   {{< tabs >}}
+   {{% tab "NPM" %}}
+
+   If using NPM, you need to make a few small changes to the code snippet from the Datadog RUM UI before pasting it into either the root `layout.tsx` or custom `_app.tsx` file (Datadog supports both):
+
+   - If your application relies on the **newer** Next.js [App Router][1] (versions 13+), paste the snippet into the root [`layout.tsx`][2] file
+   - If your Next.js application relies on the **older** Next.js [Page Router][3], paste the snippet into the custom [`_app.tsx`][4] file.
+
+   {{< code-block lang="javascript" filename="layout.tsx or _app.tsx" disable_copy="false" collapsible="true" >}}
+
+   import { datadogRum } from '@datadog/browser-rum';
+
+   datadogRum.init({
+       applicationId: '<YOUR_APPLICATION_ID>',
+       clientToken: '<CLIENT_TOKEN>',
+       site: 'datadoghq.com',
+       service:'nextjs-test',
+       env:'<ENV_NAME>',
+       // Specify a version number to identify the deployed version of your application in Datadog 
+       // version: '1.0.0', 
+       sessionSampleRate:100,
+       sessionReplaySampleRate: 20,
+       trackUserInteractions: true,
+       trackResources: true,
+       trackLongTasks: true,
+       defaultPrivacyLevel:'mask-user-input'
+   });
+       
+   datadogRum.startSessionReplayRecording();
+   {{< /code-block >}}
```

Comment:
This is an example I tested just now that sets up RUM with npm in the root layout.tsx of an app router project:

```tsx
"use client";

import Link from "next/link";
import { datadogRum } from "@datadog/browser-rum";
import { useEffect } from "react";

datadogRum.init({
  applicationId: "<YOUR_APPLICATION_ID>",
  clientToken: "<CLIENT_TOKEN>",
  site: "datadoghq.com",
  service: "next-app-router-rum-npm",
  env: "<ENV_NAME>",
  // Specify a version number to identify the deployed version of your application in Datadog
  // version: '1.0.0',
  sessionSampleRate: 100,
  sessionReplaySampleRate: 20,
  trackUserInteractions: true,
  trackResources: true,
  trackLongTasks: true,
  defaultPrivacyLevel: "mask-user-input",
});

datadogRum.startSessionReplayRecording();

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  useEffect(() => {}, []);
  return (
    <html lang="en">
      <body>
        {children}
      </body>
    </html>
  );
}
```

Reply:
We may want to add a note here like: `Because the RUM SDK needs to run on the client to collect telemetry data, the file where it is initialized via the npm package must be a client component` and link to https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs